src/JsonSchema/Constraints/Factory.php

```diff
@@ -73,9 +73,43 @@ public function __construct(
         UriRetrieverInterface $uriRetriever = null,
         $checkMode = Constraint::CHECK_MODE_NORMAL
     ) {
+        // set provided config options
+        $this->setConfig($checkMode);
+
         $this->uriRetriever = $uriRetriever ?: new UriRetriever;
         $this->schemaStorage = $schemaStorage ?: new SchemaStorage($this->uriRetriever);
-        $this->checkMode = $checkMode;
+    }
+
+    /**
+     * Set config values
+     *
+     * @param int $checkMode Set checkMode options
+     * @param bool $enable Whether to enable or disable the provided options. If null, overwrites checkMode.
+     */
+    public function setConfig($checkMode = Constraint::CHECK_MODE_NORMAL, $enable = null)
```

Comment:
`$enable` is now gone, and `setConfig()` will always overwrite. I do really want to keep the ability to enable / disable specific flags, so have added `addConfig()` and `removeConfig()` per your suggestion.